pipeline/backend/kubernetes/pod.go

```diff
@@ -107,6 +107,10 @@ func podSpec(step *types.Step, config *config) (v1.PodSpec, error) {
 		Tolerations:        tolerations(step.BackendOptions.Kubernetes.Tolerations),
 		SecurityContext:    podSecurityContext(step.BackendOptions.Kubernetes.SecurityContext, config.SecurityContext),
 	}
+
+	if (step.Type == types.StepTypeService) && (!config.StorageRwx) {
+		return spec, nil
```

Comment:
The same as with volume mounts.

Why I'm disallowed to mount my volume Read-Only?

Reply:
Maybe I'm not clear. My main idea is to create services without PVC when the storage is read-only. 
Why? This is because the next step runs in another node making it impossible to mount.

Reply:
> My main idea is to create services without PVC when the storage is read-only

I got it and asked:

> Why I'm disallowed to mount my volume in Read-Only?

[Volumes](https://woodpecker-ci.org/docs/administration/backends/kubernetes#volumes)
Why I disallowed from mounting [a secret (or configmap) with a file ca.crt of a self-hosted registry](https://github.com/woodpecker-ci/helm/pull/47) in a service, for example.

---

> the next step runs in another node making it impossible to mount

That is the cause. Seems should be fixed. But not by disallowing Read-Only mode in service.

Consider the example:
```yaml
skip_clone: true
services:
  server:
    image: alpine
    commands:
      - echo Hello from service
    ports:
      - 1234
steps:
  client:
    image: alpine
    commands:
      - echo Hello from step
  two:
    image: alpine
    commands:
      - echo Hello from step
  three:
    image: alpine
    commands:
      - echo Hello from step
```

What if all of them will be scheduled on separate node?
Why Read-Only mode matters? Why only in service?

Maybe [NodeSelector](https://woodpecker-ci.org/docs/administration/backends/kubernetes#nodeselector) will serve as a workaround.

Reply:
I got. I try to consider the mount when is declared  in the step.

Reply:
Then we should do some affinity like K3s do with their Local Storage. Haven't dug deep, though.